### PR TITLE
Feature/activity class

### PR DESCRIPTION
**Pull request by dietza**
### What does this PR do?
Finishes initial build-out structure of the Activity repository class in Activity.js.
Adds and passes tests in Activity-test.js.

### Where should the reviewer start?
Line 1 of Activity.js
Line 1 of Activity-test.js

### How is this manually tested?
run npm test in terminal to confirm that tests pass as expected.

### Any relevent tickets?
#7 #18 

### Additional considerations?
We have added issues for places to return to refactor (noted along the way, but not yet resolved before this PR)

### CODE REVIEW

File: src/Activity.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  findAllUsersAverageByDate(date, dataMetric) {
+    const dailyActivityData = this.filterDataByDate(date);
+    const allUsersDailyTotal = dailyActivityData.reduce((sum, activityInfo) => {
+      sum += activityInfo[dataMetric];
+      return sum;
+    }, 0);
+    const allUsersDailyAverage = allUsersDailyTotal / dailyActivityData.length;
+    return allUsersDailyAverage;
+  }
+}
```
Comment (DrewBradley) [inline src/Activity.js:98]: Great use of dynamic code! This will come in handing later!

File: test/Activity-test.js [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+  it('should return an array of dates a user successfully met their step goal (for all time)', () => {
+    const stepGoalSuccessDates = activity.returnSuccessfulStepDates(1, userRepo);
+    expect(stepGoalSuccessDates).to.be.an('array');
+    expect(stepGoalSuccessDates).to.deep.equal(['2019/06/17', '2019/06/20', '2019/06/22', '2019/06/23']);
+  });
+
+  it('should return the highest number of flights of stairs a user has climbed in one day', () => {
+    const stairClimbingRecord = activity.findUserStairClimbingRecord(3);
+    expect(stairClimbingRecord).to.be.an('object');
+    expect(stairClimbingRecord).to.deep.equal({ '2019/06/20': 46 });
```
Comment (DrewBradley) [inline test/Activity-test.js:382]: These variable names seem a bit vague. 🙃

File: test/Activity-test.js [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+  });
+
+  it('should calculate the average active minutes for all users on a specific date', () => {
+    const allUsersActiveMinutesAverage1 = activity.findAllUsersAverageByDate("2019/06/18", "minutesActive");
+    const allUsersActiveMinutesAverage2 = activity.findAllUsersAverageByDate("2019/06/21", "minutesActive");
+    expect(allUsersActiveMinutesAverage1).to.be.an('number');
+    expect(allUsersActiveMinutesAverage1).to.equal(185);
+    expect(allUsersActiveMinutesAverage2).to.be.an('number');
+    expect(allUsersActiveMinutesAverage2).to.equal(162.5);
+  });
```
Comment (DrewBradley) [inline test/Activity-test.js:410]: These last three tests all test the same function! Great job being thorough!

**Review by DrewBradley — APPROVED** [review]
Great work! Let's merge it!

File: src/Activity.js (hunk 1 above)
Comment (hfaerber) [reply src/Activity.js:98]: Love that you are leveraging this method to access several different data metrics depending on the argument passed.  Great job keeping this DRY

**Comment by hfaerber** [comment]
Great work you two.  I added some comments to Issue #7 as well - just sharing insights I've seen on the job around crafting user stories.   Your workflow seems to be really organized and efficient on this project.  I love that you are creating issues to go back to later when you find something that needs revisited but maybe isn't top priority at that time.  Keep it up!

**Pull request merged**